escrow/contracts/escrow.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Unlicensed
+pragma solidity ^0.8.9;
+
+import "@acala-network/contracts/token/Token.sol";
+
+contract Escrow {
+    address payable public requestor;
+    address payable public serviceProvider;
+    address public tokenAddress;
+    uint256 public amount;
+    ServiceStatus public requestorStatus;
+    ServiceStatus public serviceProviderStatus;
+
+    enum ServiceStatus {
+        Pending,
+        Confirmed,
+        Denied
+    }
+
+    constructor(
+        address _tokenAddress,
+        uint256 _amount,
+        address payable _requestor,
+        address payable _serviceProvider
+    ) {
+        requestor = _requestor;
+        serviceProvider = _serviceProvider;
+        tokenAddress = _tokenAddress;
+        amount = _amount;
+        requestorStatus = ServiceStatus.Pending;
+        serviceProviderStatus = ServiceStatus.Pending;
+    }
+
+    function requestorConfirmTaskCompletion(bool _taskConfirmation) public {
+        require(
+            msg.sender == requestor,
+            "Only the requestor can confirm his part"
```

Comment:
It is a good practice, to reformat error reposting in a way, that starts with the smart contract name. This way you know which smart contract reverted the transaction if it comes to that:
```suggestion
            "Escrow: only the requestor can confirm his part"
```